src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -34,7 +34,7 @@ internal partial class Binder
             // Create a map from type parameter name to ordinal.
             // No need to report duplicate names since duplicates
             // are reported when the type parameters are bound.
-            var names = new Dictionary<string, int>(n);
+            var names = new Dictionary<string, int>(n, StringOrdinalComparer.Instance);
```

Comment:
The default is some kind of CRC variant and is not changing (for compat reasons). It is bad enough that Dictionary has a special case that when there are massive collisions it swaps the comparer to use marvin32 and rehashes the whole table.

For our common case - "relatively short strings" - FNV should be hard to beat.